### PR TITLE
WRR-1039: Removed animation and shadow effects with cli `--no-animation` option

### PR DESCRIPTION
**Pull request by vJIYEv**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
We'd like to modify sandstone components to apply animation and shadow effects conditionally for low-performant devices.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
This PR is based on https://github.com/enactjs/cli/pull/316 and https://github.com/enactjs/cli/pull/349 which added `--no-animation` build option to cli.
Because `--no-animation` build option defines `ENACT_PACK_NO_ANIMATION` variable as `true`, I added conditions to turn off effects with `ENACT_PACK_NO_ANIMATION` variable. 

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRR-1039

### Comments
Enact-DCO-1.0-Signed-off-by: Jiye Kim (jiye.kim@lge.com)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ The following is a curated list of changes in the Enact sandstone module, newest
 
 ## [unreleased]
 
+### Changed
+
+- animation and shadow effects to be excluded with enact cli `--no-animation` build option
```
Comment (MikyungKim) [inline CHANGELOG.md:9]: I think we don't need this change log right now.

File: Button/Button.js [hunk 2]
```diff
@@ -218,6 +218,7 @@ const ButtonBase = kind({
 				collapsable,
 				collapsed,
 				roundBorder,
+				noAnimation: typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION,
```
Comment (MikyungKim) [inline Button/Button.js:221]:
```suggestion
				noAnimation: typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION,
				roundBorder,
```

File: Checkbox/Checkbox.js [hunk 3]
```diff
@@ -162,7 +163,7 @@ const CheckboxBase = kind({
 				disabled={disabled}
 				role="checkbox"
 			>
-				<div className={css.bg} />
+				<div className={classnames(css.bg, (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) ? css.noAnimation : null)} />
```
Comment (MikyungKim) [inline Checkbox/Checkbox.js:166]: I think it would be better if we define bg classnames in `compute`.
So in computed,
```
computed: {
		className: ({indeterminate, selected, standalone, styler}) => styler.append({
			selected,
			standalone,
			indeterminate,
			noAnimation: (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION)
		}),
```
and in the less file,

```
// Focus States
@checkbox-focus: {
	.checkbox-states(
		// Gathered rules into a rules-set to include the .bg selector in the resting block
		{
			...
			.bg {
				.sand-spotlight-focus-bg-colors();
			}
			&.noAnimation .bg {
				.sand-spotlight-focus-bg-colors(@shadow: {
					box-shadow: none;
				});
			}
		};
```

File: FixedPopupPanels/FixedPopupPanels.js [hunk 4]
```diff
@@ -72,6 +73,9 @@ const fixedPopupPanelsHandlers = {
  */
 const FixedPopupPanelsBase = (props) => {
 	const handlers = useHandlers(fixedPopupPanelsHandlers, props);
+	if (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) {
+		return <Viewport {...props} {...handlers} className={classnames(props.className, css.noAnimation)} noAnimation />;
+	}
 	return <Viewport {...props} {...handlers} />;
```
Comment (MikyungKim) [inline FixedPopupPanels/FixedPopupPanels.js:79]:
```suggestion
	const noAnimationProps = (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) ? {
		className: classnames(props.className, css.noAnimation),
		noAnimation: true
	} : null;

	return <Viewport {...props} {...handlers} {...noAnimationProps} />;
```

File: FlexiblePopupPanels/Panel.js [hunk 5]
```diff
@@ -208,7 +209,7 @@ const PanelBase = kind({
 						/>
 					</Cell>
 					<Cell
-						className={css.content}
+						className={classnames(css.content, (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) ? css.noAnimation : null)}
```
Comment (MikyungKim) [inline FlexiblePopupPanels/Panel.js:212]: Could we change this to use `computed` like I mentioned in `Checkbox`?

File: PageViews/PageViews.js [hunk 6]
```diff
@@ -240,7 +240,7 @@ const PageViewsBase = kind({
 					component={ViewManager}
 					duration={400}
 					index={index}
-					noAnimation={noAnimation}
+					noAnimation={(typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) || noAnimation}
```
Comment (MikyungKim) [inline PageViews/PageViews.js:243]: Let the `ENACT_PACK_NO_ANIMATION` have higher priority.

```suggestion
					noAnimation={noAnimation || (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION)}
```

File: Panels/Panels.js [hunk 7]
```diff
@@ -248,7 +248,7 @@ const PanelsBase = kind({
 					generateId={generateId}
 					id={viewportId}
 					index={index}
-					noAnimation={noAnimation}
+					noAnimation={(typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) || noAnimation}
```
Comment (MikyungKim) [inline Panels/Panels.js:251]: same here.

File: Popup/Popup.js [hunk 8]
```diff
@@ -229,7 +229,8 @@ const PopupBase = kind({
 		// If `noAlertRole` is true, alert role and aria-live will be removed. Contents of the popup won't be read automatically when opened.
 		// Otherwise, `role` will be usually `alert`.
 		role: ({noAlertRole, role}) => ((typeof role !== 'undefined') ? role : (!noAlertRole && 'alert' || null)),
-		transitionContainerClassName: ({css, position, styler}) => styler.join(css.popupTransitionContainer, position)
+		transitionContainerClassName: ({css, position, styler}) => styler.join(css.popupTransitionContainer, position),
+		noAnimation: ({noAnimation}) => (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) || noAnimation
```
Comment (MikyungKim) [inline Popup/Popup.js:233]: same here.

File: PopupTabLayout/PopupTabLayout.js [hunk 9]
```diff
@@ -259,7 +259,8 @@ const PopupTabLayoutBase = kind({
 	},
 
 	computed: {
-		className: ({collapsed, scrimType, styler}) => styler.append({collapsed}, `scrim${cap(scrimType)}`)
+		className: ({collapsed, noAnimation, scrimType, styler}) => styler.append({collapsed, noAnimation: (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) || noAnimation}, `scrim${cap(scrimType)}`),
+		noAnimation: ({noAnimation}) => (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION) || noAnimation
```
Comment (MikyungKim) [inline PopupTabLayout/PopupTabLayout.js:263]:
```suggestion
		className: ({collapsed, noAnimation, scrimType, styler}) => styler.append({collapsed, noAnimation: typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION}, `scrim${cap(scrimType)}`),
		noAnimation: ({noAnimation}) => noAnimation || (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION)
```

File: Switch/Switch.js [hunk 10]
```diff
@@ -82,7 +82,8 @@ const SwitchBase = kind({
 	computed: {
 		className: ({noAnimation, selected, styler}) => styler.append({
 			animated: !noAnimation,
-			selected
+			selected,
+			noAnimation: typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION
```
Comment (MikyungKim) [inline Switch/Switch.js:86]:
```suggestion
			noAnimation: typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION,
			selected
```

File: Popup/Popup.js (hunk 8 above)
Comment (MikyungKim) [reply Popup/Popup.js:233]: could you sort it alphabetically?

File: VideoPlayer/VideoPlayer.js [hunk 11]
```diff
@@ -2104,7 +2104,7 @@ const VideoPlayerBase = class extends Component {
 								null
 						}
 						<ControlsContainer
-							className={css.bottom + (this.state.mediaControlsVisible ? '' : ' ' + css.hidden) + (this.state.infoVisible ? ' ' + css.lift : '')}
+							className={css.bottom + (this.state.mediaControlsVisible ? '' : ' ' + css.hidden) + (this.state.infoVisible ? ' ' + css.lift : '') + (typeof ENACT_PACK_NO_ANIMATION !== 'undefined' && ENACT_PACK_NO_ANIMATION ? ' ' + css.noAnimation :  '')}
```
Comment (MikyungKim) [inline VideoPlayer/VideoPlayer.js:2107]: Can we use `classnames` instead of `+`?

**Comment by MikyungKim** [comment]
Please resolve the merge conflict. And also, would you please remove the overscrolleffect of scroller and lists? Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (vJIYEv) [reply CHANGELOG.md]: ok

File: Button/Button.js (hunk 2 above)
Comment (vJIYEv) [reply Button/Button.js]: ok

File: Checkbox/Checkbox.js (hunk 3 above)
Comment (vJIYEv) [reply Checkbox/Checkbox.js]: ok

File: FixedPopupPanels/FixedPopupPanels.js (hunk 4 above)
Comment (vJIYEv) [reply FixedPopupPanels/FixedPopupPanels.js]: ok

File: FlexiblePopupPanels/Panel.js (hunk 5 above)
Comment (vJIYEv) [reply FlexiblePopupPanels/Panel.js]: ok
I changed `FlexiblePopupPanels/Header.js` file and less file too.

File: Switch/Switch.js (hunk 10 above)
Comment (vJIYEv) [reply Switch/Switch.js]: ok

File: VideoPlayer/VideoPlayer.js (hunk 11 above)
Comment (vJIYEv) [reply VideoPlayer/VideoPlayer.js]: ok with `+` that binds classes
Is it ok with `+` for space? they are still remaining.

File: FixedPopupPanels/FixedPopupPanels.js [hunk 12]
```diff
@@ -9,6 +9,7 @@
  * @exports Header
  */
 
+import classnames from 'classnames';
```
Comment (MikyungKim) [inline FixedPopupPanels/FixedPopupPanels.js:12]: This line should go after import statement of `getTargetByDirectionFromElement`.

**Comment by MikyungKim** [comment]
After discussing with @vJIYEv , we'd like to add a global classname 'noAnimation` to remove CSS effects instead of defining it from every component.

File: FixedPopupPanels/FixedPopupPanels.js (hunk 12 above)
Comment (vJIYEv) [reply FixedPopupPanels/FixedPopupPanels.js]: this line is deleted now.

File: VideoPlayer/VideoPlayer.js [hunk 13]
```diff
@@ -7,6 +7,7 @@
  * @exports VideoPlayerBase
  */
 
+import classnames from 'classnames';
```
Comment (MikyungKim) [inline VideoPlayer/VideoPlayer.js:10]: Let's revert this change.

**Review by MikyungKim — APPROVED** [review]
LGTM
